File: system-test/sec-tests/java/Sec46TocTou.java
// SEC-46: Race Condition / TOCTOU
// Test: TC-CR-050
package com.example;

import java.nio.file.*;

public class FileDeleteService {

    public void deleteFile(String path) throws Exception {
        Path filePath = Paths.get(path);
        if (Files.exists(filePath)) {
            // Race window: attacker can swap file between check and delete
            Files.delete(filePath);
        }
    }
}

File: system-test/sec-tests/typescript/sec_46_toctou.ts
// SEC-46: Race Condition / TOCTOU
// Test: TC-CR-050
import fs from 'fs';

function deleteFile(filePath: string): void {
  if (fs.existsSync(filePath)) {
    fs.unlinkSync(filePath);  // Race window between check and use
  }
}
